### PR TITLE
Update License and add new TM InVEST logo

### LINKED ISSUE #795: Update InVEST License
Update to 2022.

I also suggest we add a "restriction" that requires derived works to carry a different name than "InVEST".

### LINKED ISSUE #1148: Handle the Trademarking of InVEST
InVEST and the design of the logo are now registered trademarks of Stanford University. We're waiting on an update from Luis from Stanford OTL in thinking about how to add the TM licenses to our software.

### LINKED ISSUE #1287: Update license to include current partners
Lisa and I noticed that our current license has an outdated set of partners listed. Specifically CAS and SRC are missing from the current list.

Related to #795 and #1148

### PR DESCRIPTION
**Pull request by dcdenu4**
## Description
Update the License from 3-Clause BSD to Apache 2.0 and add a NOTICE companion file. The custom copyright is now moved into the NOTICE file.

Update the InVEST logos to use the new registered TM logos. Add a registered TM symbol to InVEST in the Workbench.

Add an ADR file to document the decision to switch licenses.

**This review would benefit from testing the installers and Workbench**
[Windows binary](https://github.com/natcap/invest/suites/15578647243/artifacts/889035804)
[Mac binary](https://github.com/natcap/invest/suites/15578647243/artifacts/889035802)

Fixes #795 
Fixes #1287 
Fixes #1148 

## Checklist
- [x] Updated HISTORY.rst and link to any relevant issue (if these changes are user-facing)
- [ ] Updated the user's guide (if needed)
- [x] Tested the Workbench UI (if relevant)

### CODE REVIEW

File: workbench/resources/InVEST-2-256x256.ico
Comment (davemfish) [inline workbench/resources/InVEST-2-256x256.ico:1]: These ico files were just replaced with new ones right? `electron-builder-config.js` doesn't have any changes so presumably still points to these.

**Review by davemfish — CHANGES_REQUESTED** [review]
I tried this on Windows, looks great!

I wonder if you want to also distribute the new NOTICE alongside the LICENSE? Also, while we're here do you think it makes sense to move the location of our `LICENSE.txt` in the application's install directory?

Now it's at,
`<root install>/resources/LICENSE.InVEST.txt`,

whereas some other licenses are at,
`<root install>/LICENSE.electron.txt`
`<root install>/LICENSES.chromium.html`

If we wanted to move ours alongside theirs, I think we just put it in `extraFiles` as opposed to `extraResources` in `electron-builder-config.js`.

File: workbench/resources/InVEST-2-256x256.ico
Comment (dcdenu4) [reply workbench/resources/InVEST-2-256x256.ico:1]: Yep, I just wrote over the existing ones. I thought about creating new filenames to be explicit, but didn't ultimately see the big benefit.

File: doc/decision-records/ADR-0002-Switch-InVEST-License.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+Doug and James did have a discussion about whether we should reach out to prior
+contributors to get their sign off on switching licenses. Doug had noticed this
+was something other large open source projects had to contend with when 
+switching licenses. However, because the license remains permissive and the 
+fact that all major InVEST source code contributors mostly pushed changes as a
+NatCap Stanford University employee, we did not feel it necessary.
+
+## Status
+
+- PR Draft
```
[the remark this answers is not in the document]
Comment (dcdenu4) [reply doc/decision-records/ADR-0002-Switch-InVEST-License.md]: Good point, done in https://github.com/natcap/invest/pull/1385/commits/800b91cb239881f52036dbc24369a36bcef6a034

File: workbench/src/renderer/menubar/about.jsx [hunk 2]
```diff
@@ -24,51 +24,73 @@ ReactDom.render(
...
+                  >
+                    https://github.com/natcap/invest
+                  </a>
+                </td>
+              </tr>
+              <tr>
+                <td>{t('License')}</td>
+                <td>
+                  <a
+                    href="https://github.com/natcap/invest/blob/master/LICENSE.txt"
```
[the remark this answers is not in the document]
Comment (dcdenu4) [reply workbench/src/renderer/menubar/about.jsx]: Good catch. Done in https://github.com/natcap/invest/pull/1385/commits/3785889eff56c979787ea80c5b6e2f1fd2a3152c

File: workbench/src/renderer/styles/style.css [hunk 3]
```diff
@@ -73,6 +75,13 @@ body {
 	margin-bottom: -0.2rem;
 }
 
+/*registered trademark icon styling*/
+.rtm {
+	color: var(--invest-green);
+	font-size: 0.5rem;
+  margin-bottom: 0.3rem;
```
[the remark this answers is not in the document]
Comment (dcdenu4) [reply workbench/src/renderer/styles/style.css:82]: Okay, I think I replaced everything with 2 spaces... hopefully this is helpful!

**Comment by dcdenu4** [comment]
Thanks @davemfish, I think I covered everything. Bumped the distribution location of the License up one and added the Notice file.

Let me know if I helped or made more a mess with the style css indenting!

**Comment by dcdenu4** [comment]
@davemfish and @phargogh I snuck in an update to HISTORY that removes the 3.14.0 header and puts its contents in the proper spot below. This is to avoid conflict with how the auto release script updates HISTORY.

File: workbench/electron-builder-config.js [hunk 4]
```diff
@@ -30,6 +30,16 @@ const config = {
...
+  extraFiles: [
+    {
+      from: '../LICENSE.txt',
+      to: 'LICENSE.InVEST.txt',
+    },
+    {
+      from: '../NOTICE.txt',
+      to: 'NOTICE.InVEST.txt',
+    },
+  ]
```
Comment (davemfish) [inline workbench/electron-builder-config.js:42]: missing comma here

**Review by davemfish — CHANGES_REQUESTED** [review]
Thanks @dcdenu4 , just a syntax error in the electron builder config.

**Review by phargogh — CHANGES_REQUESTED** [review]
Just one small change and a question about the status, but it looks great otherwise!  I tried out the mac build of the workbench and all of the updates were looking really good.

File: workbench/electron-builder-config.js (hunk 4 above)
Comment (dcdenu4) [reply workbench/electron-builder-config.js]: Good eye! https://github.com/natcap/invest/pull/1385/commits/0d5adc586a10e4fdf57259c6bdfff5c1eb844cbd

File: doc/decision-records/ADR-0002-Switch-InVEST-License.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+Doug and James did have a discussion about whether we should reach out to prior
+contributors to get their sign off on switching licenses. Doug had noticed this
+was something other large open source projects had to contend with when 
+switching licenses. However, because the license remains permissive and the 
+fact that all major InVEST source code contributors mostly pushed changes as a
+NatCap Stanford University employee, we did not feel it necessary.
+
+## Status
+
+Complete
```
[the remark this answers is not in the document]
Comment (dcdenu4) [reply doc/decision-records/ADR-0002-Switch-InVEST-License.md]: Added to this noting the 3.14.0 release. I think generally the "Status" field is maybe only applicable when the ADR is created before and separate of the implementation. I almost think that "Status" or "Details" could live at the top of the ADR... things to think about and iterate on!

**Review by phargogh — APPROVED** [review]

**Review by davemfish — APPROVED** [review]

**Pull request merged**